src/object_store/utils.rs

```diff
@@ -44,3 +48,33 @@ pub async fn fast_upload(from: &StdPath, to: String) -> object_store::Result<(),
         Ok(())
     }
 }
+
+// Go through all known keys for object store and convert them to corresponding file_io ones.
+//
+// For now only converts S3 keys.
+// TODO: At some point this should be redundant, since there is an OpenDAL adapter for object_store,
+// https://github.com/apache/iceberg-rust/issues/172
+pub fn object_store_opts_to_file_io_props(
+    opts: &HashMap<String, String>,
+) -> HashMap<String, String> {
+    let mut props = HashMap::new();
+
+    for (key, val) in opts.iter() {
+        let key = match AmazonS3ConfigKey::from_str(key) {
+            Ok(AmazonS3ConfigKey::AccessKeyId) => S3_ACCESS_KEY_ID,
+            Ok(AmazonS3ConfigKey::SecretAccessKey) => S3_SECRET_ACCESS_KEY,
+            Ok(AmazonS3ConfigKey::Region) => S3_REGION,
+            Ok(AmazonS3ConfigKey::Endpoint) => S3_ENDPOINT,
+            _ => key, // for now just propagate any non-matched keys
+        };
+
+        props.insert(key.to_string(), val.clone());
+    }
+
+    // FileIO requires the region prop even when the S3 store doesn't (e.g. MinIO)
+    props
+        .entry(S3_REGION.to_string())
+        .or_insert("us-east-1".to_string());
```

Comment:
Do you think we should use a dummy value like "dummy" so that it fails when using real AWS S3 (rather than using real AWS S3 with an incorrect region)?

Reply:
That's a good point, doing that.